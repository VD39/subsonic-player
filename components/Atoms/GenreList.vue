<script setup lang="ts">
defineProps<{
  genres: Genre[];
}>();
</script>

<template>
  <ul class="commaList">
    <li
      v-for="genre in genres"
      :key="genre.name"
      data-test-id="genre-list-item"
    >
      <NuxtLink
        :aria-label="`View all albums with ${genre.name} genre`"
        class="link"
        :to="`/genre/albums/${encodeURIComponent(genre.name)}`"
      >
        {{ genre.name }}
      </NuxtLink>
    </li>
  </ul>
</template>
