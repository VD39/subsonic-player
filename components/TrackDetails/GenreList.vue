<script setup lang="ts">
defineProps<{
  genres: Genre[];
}>();
</script>

<template>
  <ul class="commaList">
    <li
      v-for="genre in genres"
      :key="genre.name"
      data-test-id="genre-list-item"
    >
      <NuxtLink :to="`/genre/albums/${genre.name}`">
        {{ genre.name }}
      </NuxtLink>
    </li>
  </ul>
</template>
