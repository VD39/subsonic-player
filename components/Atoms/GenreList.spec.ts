import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import GenreList from './GenreList.vue';

function factory(props = {}) {
  return mount(GenreList, {
    props: {
      genres: [
        {
          name: 'name',
        },
        {
          name: 'name1',
        },
      ],
      ...props,
    },
  });
}

describe('GenreList', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows the correct number of genre content', () => {
    expect(wrapper.findAll('[data-test-id="genre-list-item"]').length).toBe(2);
  });
});
