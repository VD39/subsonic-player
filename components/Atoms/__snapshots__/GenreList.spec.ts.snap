// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GenreList > matches the snapshot 1`] = `
"<ul class="commaList">
  <li data-test-id="genre-list-item"><a aria-label="View all albums with name genre" class="link">name</a></li>
  <li data-test-id="genre-list-item"><a aria-label="View all albums with name1 genre" class="link">name1</a></li>
</ul>"
`;
